### PR TITLE
[FIX] sale: only try to split if name is set

### PR DESCRIPTION
**Pull request by wtaferner**
**Description of the issue/feature this PR addresses:**
In case the name is not set it will fail to retrieve the wanted behavior

**Current behavior before PR:**
Error

**Desired behavior after PR is merged:**
No Error

@nim-odoo 
May I ask you to merge? Should be straight forward.

Info: @wt-io-it

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/models/sale.py [hunk 1]
```diff
@@ -1489,7 +1489,7 @@ def product_uom_change(self):
     def name_get(self):
         result = []
         for so_line in self.sudo():
-            name = '%s - %s' % (so_line.order_id.name, so_line.name.split('\n')[0] or so_line.product_id.name)
+            name = '%s - %s' % (so_line.order_id.name, so_line.name and so_line.name.split('\n')[0] or so_line.product_id.name)
```
Comment (fmdl) [inline addons/sale/models/sale.py]: Hi @wtaferner I ave one remark :
- I think it is better to use `display_name` indeed `name` of product `or so_line.product_id.name` ->` or so_line.product_id.display_name`

Comment (wtaferner) [reply addons/sale/models/sale.py]: @fmdl 
It is a bugfix, so it would be the call of @nim-odoo, but it makes sense somehow what you suggest.

**Comment by mart-e** [comment]
Hi,

The name is a required field. How do you call the name_get without a value?

**Comment by wtaferner** [comment]
@mart-e 
I guess in the onchange without the webclient in a new record.
I did not dig deeper as it was obvious for me where to fix this minor issue and as you see the name is set automatically, so why should I put it an empty string if I want the onchange behavior?. 

Should I close it or do you think it is worth it to be explicit before applying some methods to a string which is expected to be like it b/c it is required in the database?

**Comment by mart-e** [comment]
Well I still would like to know how to reproduce it. The thing is that theoretically we could get this everywhere. You are changing this way:

```python
'%s - %s' % (so_line.order_id.name, so_line.name and so_line.name[...])
```
but why couldn't we get `False` on the order name as well and need to change for:
```python
'%s - %s' % (so_line.order_id.name or '', so_line.name and so_line.name[...])
```
I just made a try of adding the field display_name in the embedded tree view of sale.order.line in a sale.order but the field was correctly computed only when saving the record (with the required value set).

**Comment by wtaferner** [comment]
@mart-e 
As soon as I find the time I will try to revert my fix and dig deeper. And yes, I think we need to check for certain application of methods if we have the proper variable type in hands or at least avoid throwing exceptions. Unfortunately we have no explicit parameter and return types defined yet, otherwise we would probably have no issue as `so_line.name` would be an empty `string` instead of a `boolean`.

**Comment by wtaferner** [comment]
Fixed in https://github.com/odoo/odoo/commit/a9cae4d3a7163c817f612c12aabc2e5b7a3af900

**Pull request closed without merging**